### ISSUE TITLE
Cannot build PCL against VTK with OpenGL2

### ISSUE THREAD

**Issue by VictorLamoine** [PointCloudLibrary/pcl#1275]
How to reproduce
---
- Compile / install VTK with `VTK_RENDERING_BACKEND = OpenGL2`
- Compile PCL against this VTK version

Error
---
```
pcl/src/visualization/src/pcl_visualizer.cpp:87:38: fatal error: vtkOpenGLHardwareSupport.h: Aucun fichier ou dossier de ce type
 #include <vtkOpenGLHardwareSupport.h>
```

https://github.com/Kitware/VTK/commit/249fd506f040e41d952be57b092da94c66bc4662

Note: Close #1234 when this is solved!
Note: Revert https://github.com/PointCloudLibrary/pcl/pull/1533 when this issue is closed!

**Comment by VictorLamoine** [comment]
```
pcl/src/visualization/src/pcl_visualizer.cpp:87:38: fatal error: vtkOpenGLHardwareSupport.h: Aucun fichier ou dossier de ce type
 #include <vtkOpenGLHardwareSupport.h>
```

**Comment by VictorLamoine** [comment]
I created a big mesh with blender (8060918 points)

VTK + OpenGL backend: less than 5 fps
VTK + OpenGL2 backend: more than 60 fps

Note that OpenGL2 is now **default** in VTK trunk!
Given the outrageous performance improvement, we really should work on making PCL compatible with VTK+OpenGL2.

Help is appreciated because I'm not quite sure how to modify PCL code using OpenGL.

**Comment by taketwo** [comment]
Sounds interesting. I'll try to have a look, but no guarantees...

**Comment by VictorLamoine** [comment]
I asked the VTK community:
http://vtk.1045678.n5.nabble.com/Migrating-from-OpenGL-to-OpenGL2-td5735123.html

I'll give it a try a little bit later but it doesn't look like an easy fix to apply!

**Comment by mcleary** [comment]
There are plans to support VTK OpenGL2 backend?

**Comment by VictorLamoine** [comment]
The issue if here just for that, unfortunately what I tried did not work a the moment.
If you have time and skills to make the changes, pull requests are welcome!

**Comment by mcleary** [comment]
@VictorLamoine, I understand, I asked to see if there is anyone already working on this. I'll give it a try.

**Comment by bchretien** [comment]
:+1:  for this, we have a similar issue on Arch Linux, where VTK 7.0 is now used with the OpenGL2 backend.

**Comment by taketwo** [comment]
Fixed with #1534.
